### PR TITLE
Add review features

### PR DESCRIPTION
**Pull request by cyredhat**

### CODE REVIEW

File: app/reviews/components/ReviewStarInput.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    register,
+    formState: { errors },
+  } = useFormContext()
+  const name = `rating-${orderItem.productId}`
+  let stars: boolean[] = [true, true, true, true, true]
+  ;(() => {
+    for (let i = 0; i < 5; i++) {
+      stars[i] = i + 1 <= rating
+    }
+  })()
```
Comment (paphonb) [inline app/reviews/components/ReviewStarInput.tsx:24]: Why do we need an IIFE here?

File: app/reviews/components/ReviewStarInput.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  console.log(errors)
+
+  const error = Array.isArray(errors[name])
+    ? errors[name].join(', ')
+    : errors[name]?.message || errors[name]
+  console.log(error)
```
Comment (paphonb) [inline app/reviews/components/ReviewStarInput.tsx:30]: Remove console log

**Review by komson-lim — APPROVED** [review]

File: app/reviews/components/ReviewStarInput.tsx [hunk 3]
```diff
@@ -17,11 +17,12 @@ export function ReviewStarInput({ orderItem }: ReviewStarInputProps) {
...
   const name = `rating-${orderItem.productId}`
   let stars: boolean[] = [true, true, true, true, true]
-  ;(() => {
+  const update = () => {
     for (let i = 0; i < 5; i++) {
       stars[i] = i + 1 <= rating
     }
-  })()
+  }
+  update()
```
Comment (paphonb) [inline app/reviews/components/ReviewStarInput.tsx:25]: I mean can't we just execute the for loop directly without a function

Comment (cyredhat) [reply app/reviews/components/ReviewStarInput.tsx:25]: True

**Comment by cyredhat** [comment]
1. review button when order status is 'COMPLETED'
2. review page
3. 'REVIEWED' order status
4. reviews in product detail page

image
image
image
image
image
image
image

**Review by paphonb — APPROVED** [review]

**Pull request merged**